src/monitor-plan-workspace/monitor-plan.repository.ts

```diff
@@ -61,11 +87,49 @@ export class MonitorPlanWorkspaceRepository extends Repository<MonitorPlan> {
     }
   }
 
-  async getMonitorPlan(planId: string): Promise<MonitorPlan> {
-    return this.createQueryBuilder('plan')
-      .innerJoinAndSelect('plan.plant', 'plant')
-      .where('plan.id = :planId', { planId })
-      .getOne();
+  async getMonitorPlan(planId: string, full = false): Promise<MonitorPlan> {
+    return this.findOne({
+      where: { id: planId },
+      relations: {
+        plant: true,
+        ...(full
+          ? {
+              beginReportingPeriod: true,
+              endReportingPeriod: true,
+              locations: {
+                stackPipe: true,
+                unit: true,
+              },
+              reportingFrequencies: {
+                beginReportingPeriod: true,
+                endReportingPeriod: true,
+              },
+            }
+          : {}),
+      },
+    });
+  }
+
+  async getMonitorPlanUnitStackConfigs(planId: string) {
+    const configIds = (
+      await this.query(
+        `
+      SELECT config_id FROM camdecmpswks.unit_stack_configuration usc
```

Comment:
As discussed, this logic has been updated to avoid using this particular view, and to instead get a plan's unit stack configurations based on its locations and begin and end dates. This logic has been moved to `getUnitStackConfigsByMonitorPlanId` in `UnitStackConfigurationRepository` and `UnitStackConfigurationWorkspaceRepository`.